Had to modify direction of bias with additive bias, as the formula 7.…

CYShapland opened the pull request:
…4 is beta-bias. Therefore is the bias should be positive if the study favours comparators, therefore bias needs to be positive. And bias negative to shift the estimate to the right if study favours experimental.

CYShapland committed:
Had to modify direction of bias with additive bias, as the formula 7.4 is beta-bias. Therefore is the bias should be positive if the study favours comparators, therefore bias needs to be positive. And bias negative to shift the estimate to the right if study favours experimental.

mcguinlu commented on R/append_values.R:
```diff
@@ -36,89 +36,29 @@ tri_append_bias <- function(dat, values, common = T) {
-#'
-#' @export
-
-tri_append_indirect <- function(dat, values, common = T) {
```
As this is a package, and other users may want to make use of the indirectness aspect, I would like to keep the `tri_append_indirect` function in this file.

mcguinlu commented:
Hey @CYShapland, thanks for the PR! 

> And bias negative to shift the estimate to the right if study favours experimental

I think we may have our wires crossed here. The directions (left/right) used in the `tri_append_bias()` function (shown below) are the absolute directions of bias, not the direction in which we want to shift the effect estimate. https://github.com/mcguinlu/triangulate/blob/1b2872c0bb3fe75b7091e1f747bea468977b3948/R/append_values.R#L116-L121

For example, in Figure 7.1, the absolute direction of bias is "right" when the assessment "Favours comparator": 

![exampleDirection](https://github.com/mcguinlu/triangulate/assets/41202585/4729ecf5-76a4-475a-bf2f-09159857ed9a)

When defining the sign in `tri_append_bias()`, if the bias is expected to pull the effect to the _**right**_ (absolute direction of bias = right / "Favours comparator"), it is given a positive sign. This is because (as you point out) in Eq 7.4, we are substracting the total additive bias from the effect estimate $y_i$. 

So to correct for an absolute direction of bias to the _**right**_, we want the sign of the bias correction to be positive (e.g. _N(0.9, 0.5)_), so that when it is subtracted from the effect estimate, the overal impact is negative (and so we shift the effect estimate to the left):

$y_i - (0.6)$ = $y_i - 0.6$

Conversely, when the assessment is "Favours experimental", the absolute direction of bias is to the _**left**_, and we want the sign of the bias correction to be negative (e.g. _N(-0.9, 0.5)_), so that we are shifting the effect estimate to the right in Eq 7.4:

$y_i - (-0.6)$ = $y_i + 0.6$

So I think the original coding (where the bias is mutiplied by -1 when absolute direction of bias is "left") is correct. Does that make sense to you? It took me ages to get my head around this, so please let me know if you think I've still got it backwards.

Commit:
Fix error in conversion to absolute direction

mcguinlu commented:
And just to illustrate that I find it hard to keep it straight in my head, I actually made a mistake in the `tri_absolute_direction()` function. 

```
.$t == "add" & d == "favours comparator" ~ "left",
.$t == "add" & d == "favours experimental" ~ "right",
```

should be (as per Figure 7.1)

```
.$t == "add" & d == "favours comparator" ~ "right",
.$t == "add" & d == "favours experimental" ~ "left",
```

I've fixed this in this PR now, so I would suggest you pull down the changes and re-run your code to make sure it's working as expected!

Commit:
Move to new GHA R CMD check

mcguinlu commented on R/append_values.R:
```diff
@@ -7,7 +7,7 @@
 #' @export
 
-tri_append_bias <- function(dat, values, common = T) {
+tri_append_bias_modi <- function(dat, values, common = T) {
```
I think we should leave the name the same, as this is an issue that will need to be resolved for all users of the package.

Commit:
Reorder columns in example datasets, and clean up documentation for these datasets

Commit:
Add/update long form documentation and articles

Commit:
Add empty citation file

Commit:
Update Github workflows

Commit:
Update gitignore

Commit:
Update DESC

Commit:
UNdo function name change and re-add indirectness code

Commit:
Fix error/order of abosulte direction computation

Commit:
Fix study id mismatch between the two datasets

Commit:
Fix merge issues

Merge commit '5ef1ab276494e98df7c854e746b73ca27089864c'

#Conflicts:
#	.github/workflows/R-CMD-check.yaml
#	README.Rmd
#	README.md
#	man/figures/README-unnamed-chunk-8-1.png
